### PR TITLE
tiledbsoma 1.15.0 pre-check

### PR DESCRIPTION
**Pull request by johnkerl**
Following our [established procedure](https://github.com/single-cell-data/TileDB-SOMA/wiki/Branches-and-releases)

### CODE REVIEW

File: recipe/meta.yaml
Comment (jdblischak) [inline recipe/meta.yaml:51]: This needs to be `- tiledb >=2.27.0,<2.28` with no selectors

**Review by jdblischak — CHANGES_REQUESTED** [review]
Adding back `tiledb` to the host requirements of libtiledbsoma should fix some of the compilation errors

File: recipe/meta.yaml
Comment (jdblischak) [inline recipe/meta.yaml:49]:
```suggestion
        - tiledb >=2.27.0,<2.28
```

No selectors. Those were only for the osx-64 workaround

File: recipe/meta.yaml
Comment (jdblischak) [inline recipe/meta.yaml:96]:
```suggestion
        - scipy <1.15.0
```

Not currently a problem, but this will avoid future problems just like upstream (xref: https://github.com/single-cell-data/TileDB-SOMA/pull/3445)

File: recipe/meta.yaml
Comment (jdblischak) [inline recipe/meta.yaml:98]: TileDB-Py is an optional dependency now, right? I recommend moving this requirement to the test section. That way we can still confirm it can be co-installed with the correct version of tiledb-py, but allow users the possibility to install a minimal tiledbsoma-py conda env.

**Review by jdblischak — CHANGES_REQUESTED** [review]
Mostly looking good now. Just a few minor suggestions to improve the TileDB-Py requirements

File: recipe/meta.yaml
Comment (johnkerl) [reply recipe/meta.yaml:98]: Thanks @jdblischak !

Let me 'get this green' for pre-check & unblock some things in the [dependency chain](https://github.com/single-cell-data/TileDB-SOMA/wiki/Branches-and-releases) tonight & I'll definitely do that moving forward

File: recipe/meta.yaml
Comment (johnkerl) [reply recipe/meta.yaml:96]: Thanks @jdblischak ! I'll apply this mod, too, for the 'real' feedstock PR coming up after this pre-check passes (hopefully tonight!)

**Comment by johnkerl** [comment]
`osx_64` passed on retry

**Comment by johnkerl** [comment]
> Let me 'get this green' for pre-check & unblock some things in the [dependency chain](https://github.com/single-cell-data/TileDB-SOMA/wiki/Branches-and-releases) tonight & I'll definitely do that moving forward

Derp, I needed to do this too
https://github.com/single-cell-data/TileDB-SOMA/pull/3460
-- so I'll re-run this pre-check with (a) that PR once backported to `release-1.15` and merged; (b) `somacore==1.0.26` here; (c) `scipy<0.15` here; (d) tiledb-py moved to `tests` as long as I can figure it out

**Comment by johnkerl** [comment]
@jdblischak https://github.com/TileDB-Inc/tiledbsoma-feedstock/pull/237/commits/7cb1d9f8f410574e42612b5749e2b9dd456bdfd1 has the aformentioned four changes

**Comment by johnkerl** [comment]
This commit
https://github.com/TileDB-Inc/tiledbsoma-feedstock/pull/237/commits/721f2fce128bd4136d75f47cac210de8c5d54830
is because 45 minutes (!) into the build on the previous commit I got a failure with

```
2024-12-17T05:50:07.1157920Z Encountered problems while solving:
2024-12-17T05:50:07.1158670Z   - nothing provides scipy <0.15.0 needed by tiledbsoma-py-1.15.0-py310h06877d3_0
2024-12-17T05:50:07.1159270Z   - unsupported request
```

which I cannot with.

**Comment by jdblischak** [comment]
> - nothing provides scipy <0.15.0 needed by tiledbsoma-py-1.15.0-py310h06877d3_0

This is bizarre. Clearly scipy <0.15.0 exists. I can investigate later today, but I agree let's not block the release for this

> is because 45 minutes (!) into the build on the previous commit I got a failure with

This delay is because scipy is only a runtime requirement. Thus this didn't surface until the test section when the runtime dependencies are installed.

**Comment by jdblischak** [comment]
> - nothing provides scipy <0.15.0 needed by tiledbsoma-py-1.15.0-py310h06877d3_0

Figured it out. It's scipy 1.15.0, not 0.15.0

File: recipe/meta.yaml
Comment (jdblischak) [inline recipe/meta.yaml:97]:
```suggestion
        - scipy <1.15.0
```

Comment (johnkerl) [reply recipe/meta.yaml:97]: @jdblischak thanks for catcthing the typo.

I still fear it won't let me pin under something that doesn't exist _yet_.

I'll spend another 45-minute run trying that ...

**Review by jdblischak — COMMENTED** [review]
Pre-check looks good! 🚀

**Pull request closed without merging**
